app/src/main/java/com/seafile/seadroid2/transfer/UploadTask.java

```diff
@@ -80,14 +87,26 @@ public boolean isCancelled() {
                 }
             };
             if (isUpdate) {
-                dataManager.updateFile(repoName, repoID, dir, path, monitor, isCopyToLocal);
+                if (byBlock) {
+                    dataManager.updateByBlocks(repoName, repoID, dir, path, monitor, isCopyToLocal, version);
+                } else {
+                    dataManager.updateFile(repoName, repoID, dir, path, monitor, isCopyToLocal);
+                }
             } else {
-                Log.d(DEBUG_TAG, "Upload path: " + path);
-                dataManager.uploadFile(repoName, repoID, dir, path, monitor, isCopyToLocal);
+                if (byBlock) {
+                    dataManager.uploadByBlocks(repoName, repoID, dir, path, monitor, isCopyToLocal, version);
+                } else {
+                    dataManager.uploadFile(repoName, repoID, dir, path, monitor, isCopyToLocal);
+                }
```

Comment:
There is a lot of code duplication thoughout this patch

- upload file (unencrypted, old method)
- upload by block (encrypted)

Is there a reason to keep the old method? why not just use "upload/download-by-block" for all file transfer (including unencrypted)?